daemon/main.go

```diff
@@ -434,22 +437,33 @@ func initEnv() {
 }
 
 func runDaemon() {
-	if consulAddr != "" {
-		consulDefaultAPI := consulAPI.DefaultConfig()
-		consulSplitAddr := strings.Split(consulAddr, "://")
-		if len(consulSplitAddr) == 2 {
-			consulAddr = consulSplitAddr[1]
-		} else if len(consulSplitAddr) == 1 {
-			consulAddr = consulSplitAddr[0]
+	switch kvStore {
+	case kvstore.Consul:
+		if consulAddr != "" {
+			consulDefaultAPI := consulAPI.DefaultConfig()
+			consulSplitAddr := strings.Split(consulAddr, "://")
+			if len(consulSplitAddr) == 2 {
+				consulAddr = consulSplitAddr[1]
+			} else if len(consulSplitAddr) == 1 {
+				consulAddr = consulSplitAddr[0]
+			}
+			consulDefaultAPI.Address = consulAddr
+			config.ConsulConfig = consulDefaultAPI
 		}
-		consulDefaultAPI.Address = consulAddr
-		config.ConsulConfig = consulDefaultAPI
-	}
-	if len(etcdAddr) != 0 && config.EtcdCfgPath == "" {
-		config.EtcdConfig = &etcdAPI.Config{}
-		config.EtcdConfig.Endpoints = etcdAddr
+	case kvstore.Etcd:
+		if len(etcdAddr) != 0 && config.EtcdCfgPath == "" {
+			config.EtcdConfig = &etcdAPI.Config{}
+			config.EtcdConfig.Endpoints = etcdAddr
+		}
+	case kvstore.Local:
+		log.Infof("Using local storage for key-value store")
+	default:
+		log.Warningf("unsupported key-value store provided; using local storage for key-value store")
```

Comment:
I think we should handle "no option" specified differently from an unknown kvstore type specified. Unknown should error out. No option should default to local storage without a warning of an unknown kvstore type